examples/generative/corrdiff/tests/test_dataset.py

```diff
@@ -14,55 +14,69 @@
 # # See the License for the specific language governing permissions and
 # # limitations under the License.
 
-# from training.dataset import _ZarrDataset, FilterTime
-# import torch
-# import datetime
-# import numpy as np
-# import os
+from datasets.cwb import (
+    _ZarrDataset,
+    get_zarr_dataset,
+)
+import torch
+import datetime
+import numpy as np
+import os
+import joblib
 
 
-# import pytest
+import pytest
 
-# path = "/lustre/fsw/sw_climate_fno/nbrenowitz/2023-01-24-cwb-4years.zarr"
+path = os.getenv(
+    "CWB_DATA_PATH", "/lustre/fsw/sw_climate_fno/nbrenowitz/2023-01-24-cwb-4years.zarr"
+)
 
 
-# @pytest.fixture
-# def dataset():
-#     if not os.path.isdir(path):
-#         pytest.skip()
-#     else:
-#         return _ZarrDataset(path)
+@pytest.fixture
+def dataset():
+    if not os.path.isdir(path):
+        pytest.skip()
+    else:
+        return _ZarrDataset(path)
 
 
-# def test_zarr_dataset(dataset):
-#     dataset[0]
-#     assert len(dataset) > 0
-#     assert dataset[len(dataset) - 1]
+def test_zarr_dataset(dataset):
+    dataset[0]
+    assert len(dataset) > 0
+    assert dataset[len(dataset) - 1]
 
 
-# def test_zarr_dataset_channels(dataset):
-#     assert dataset.input_channels()
-#     assert dataset.output_channels()
+def test_zarr_dataset_channels(dataset):
+    assert dataset.input_channels()
+    assert dataset.output_channels()
 
 
-# def test_zarr_dataset_time(dataset):
-#     isinstance(dataset.time()[0].year, int)
+def test_zarr_dataset_time(dataset):
+    isinstance(dataset.time()[0].year, int)
 
 
-# def test_zarr_dataset_get_valid_time_index(dataset):
-#     ans = dataset._get_valid_time_index(0)
-#     assert isinstance(ans, np.int64)
+def test_zarr_dataset_get_valid_time_index(dataset):
+    ans = dataset._get_valid_time_index(0)
+    assert isinstance(ans, np.int64)
 
-# def test_filter_time():
-#     class MockData(torch.utils.data.Dataset):
 
-#         def __getitem__(self, idx):
-#             return self.time()[idx]
+def test_train_test_split():
+    ds = get_zarr_dataset(data_path=path, train=True)
+    assert not any(t.year == 2021 for t in ds.time())
```

Comment:
I'm not sure about this test... What if we want to train on 2021?

Reply:
The 2021 hardcodes already existed so not new in the PR. I just moved the harcodes to new places. Would prefer to not remove rn, but could parameterize it in a future PR.